### DISCUSSION TITLE
Is there a way to make `@XmlElement(true)` the default?

### DISCUSSION THREAD

**Discussion by sschuberth** [pdvrieze/xmlutil#202] [Q&A] [answered]
I have to deal with an XML file that never uses attributes but encodes even primitive types as tags, so I end up with a lot of `@XmlElement(true)` annotations next to my properties. Is there a way, e.g. via a policy, to make "deserialize as tag" the default behavior?

**Comment by pdvrieze** [comment, marked as answer]
Hello @sschuberth, yes there is. There are different approaches, but [`defaultPrimitiveOutputKind`](https://pdvrieze.github.io/xmlutil/serialization/nl.adaptivity.xmlutil.serialization/-xml-serialization-policy/index.html#-1771267375%2FFunctions%2F85105109) should do this. Otherwise, override the policy with, [`effectiveOutputKind`](https://pdvrieze.github.io/xmlutil/serialization/nl.adaptivity.xmlutil.serialization/-xml-serialization-policy/index.html#-1352597641%2FFunctions%2F85105109), or [`defaultOutputKind`](https://pdvrieze.github.io/xmlutil/serialization/nl.adaptivity.xmlutil.serialization/-xml-serialization-policy/index.html#-1771267375%2FFunctions%2F85105109)

**Reply by sschuberth** [reply]
Thanks, using

```
val xml = XML {
    policy = object : DefaultXmlSerializationPolicy({ pedantic = false }) {
        @ExperimentalXmlUtilApi
        override fun defaultOutputKind(serialKind: SerialKind) = OutputKind.Element
    }
}
```

works for me. I chose that approach over setting `defaultPrimitiveOutputKind` to better document in code that *any* primitive / object is encoded as an element.

**Reply by pdvrieze** [reply]
I'm happy the policy allows you to do what you intend (and still allows for overrides in this case by using the annotations)
